public/app/core/components/AccessControl/Permissions.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+
+import { Button } from '@grafana/ui';
+import { SlideDown } from 'app/core/components/Animations/SlideDown';
+import { AddPermission } from './AddPermission';
+import { PermissionList } from './PermissionList';
+import { PermissionTarget, ResourcePermission, SetPermission, Description } from './types';
+
+const EMPTY_PERMISSION = '';
+
+const INITIAL_DESCRIPTION: Description = {
+  permissions: [],
+  assignments: {
+    teams: false,
+    users: false,
+    builtInRoles: false,
+  },
+};
+
+export type Props = {
+  resource: string;
+  resourceId: number;
+
+  canListUsers: boolean;
+  canSetPermissions: boolean;
+};
+
+export const Permissions = ({ resource, resourceId, canListUsers, canSetPermissions }: Props) => {
+  const [isAdding, setIsAdding] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isAdding, setIsAdding] = useState(false);
```
Type `boolean` is inferred